### PR TITLE
Restyle mobile nav trigger and address bugs related to its "stickiness"

### PR DESCRIPTION
**Pull request by erikphansen**
When this is merged and formation is published to npm, we can merge https://github.com/department-of-veterans-affairs/vets-website/pull/15711

## Description
This restyles the appearance of the mobile sub-navigation trigger button (https://github.com/department-of-veterans-affairs/va.gov-team/issues/13325) and fixes a handful of bugs (https://github.com/department-of-veterans-affairs/va.gov-team/issues/8256).

## Testing done
Local

## Screenshots

https://user-images.githubusercontent.com/20728956/105220039-01a11080-5b0c-11eb-9231-16f87a7100d4.mp4

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: packages/formation/js/sidenav.js
Comment (erikphansen) [inline packages/formation/js/sidenav.js:44]: this was never used

File: packages/formation/js/sidenav.js
Comment (erikphansen) [inline packages/formation/js/sidenav.js:100]: We are no longer hiding the trigger button when the menu is open. This may have been done because of a layering issue; the trigger appeared in front of the menu itself. But who knows.

File: packages/formation/js/sidenav.js
Comment (erikphansen) [inline packages/formation/js/sidenav.js:105]: I see no reason to conditionally move focus to the menu's "close" button as soon as it opens.

File: packages/formation/sass/modules/_m-nav-sidebar.scss
Comment (erikphansen) [inline packages/formation/sass/modules/_m-nav-sidebar.scss:375]: all styling is in `_m-button.scss` now

File: packages/formation/sass/modules/_m-button.scss
Comment (erikphansen) [inline packages/formation/sass/modules/_m-button.scss:132]: Once the position of the trigger gets `fixed` its z-index comes into play and we need to make sure that it's _behind_ the sidenav menu itself.

File: packages/formation/__tests__/js/sidenav/sidenav.test.js
Comment (erikphansen) [inline packages/formation/__tests__/js/sidenav/sidenav.test.js:107]: we are no longer hiding the trigger button when we open the menu

**Review by ncksllvn — APPROVED** [review]
LGTM!

**Review by kelsonic — APPROVED** [review]
LGTM! 🎉

File: packages/formation/__tests__/js/sidenav/sidenav.test.js
Comment (erikphansen) [reply packages/formation/__tests__/js/sidenav/sidenav.test.js:13]: Yes that's the "hamburger" icon. Apparently we can't use Fontawesome markup/classes here so I'm using the SVG directly.

Comment (ncksllvn) [reply packages/formation/__tests__/js/sidenav/sidenav.test.js:13]: Any way we can use a veggie burger icon instead?

Comment (erikphansen) [reply packages/formation/__tests__/js/sidenav/sidenav.test.js:13]: Maybe. But I don't want to wait for approval from design for that change.

Comment (cvalarida) [reply packages/formation/__tests__/js/sidenav/sidenav.test.js:13]: I only approve vegan kebabs.

File: packages/formation/sass/modules/_m-button.scss
Comment (cvalarida) [inline packages/formation/sass/modules/_m-button.scss:151]: This is 100% something I should know but don't: Is this duplicating any existing styling we've got? I'd love to trim down our styles if we can.

File: packages/formation/sass/modules/_m-nav-sidebar.scss
Comment (cvalarida) [inline packages/formation/sass/modules/_m-nav-sidebar.scss:375]: Why move this into `_m-buttons.scss`?

Without looking, my guess is that `_m-nav-sidebar.scss` is only included in `static-pages.entry.css`, and not in the application bundles, whereas `_m-button.scss` is included in both. If only the static pages use this styling, we probably shouldn't include it in the application bundles.

Mind you, once the design system team [culls the unused styles](https://github.com/department-of-veterans-affairs/va.gov-team/issues/17593#issuecomment-748370271) during the application and content builds, this will be less meaningful, but I don't know when that's going to happen yet.

At any rate, if I were to look for this `sidebarnav` class, I'd look in `_m-nav-sidebar.scss`. :man_shrugging:

**Review by cvalarida — COMMENTED** [review]
The bug fixes look good to me, but I'm holding off on my approval until I hear back from @rtwell on whether we want to make the style change in our design system.

**Comment by rtwell** [comment]
looking good to me, tyvm for the hard work 💯

**Review by cvalarida — APPROVED** [review]
Awesome. Thanks for checking, @rtwell! :shipit:

File: packages/formation/sass/modules/_m-nav-sidebar.scss
Comment (erikphansen) [reply packages/formation/sass/modules/_m-nav-sidebar.scss:375]: Good point. We already had some styling for `.va-btn-sidebarnav-trigger` in `_m-buttons.scss` so I flipped a coin and decided to consolidate things into that file. But perhaps I should move all that stuff into this file instead?

File: packages/formation/sass/modules/_m-button.scss
Comment (erikphansen) [reply packages/formation/sass/modules/_m-button.scss:151]: We have styles similar to this used in https://github.com/department-of-veterans-affairs/vets-website/blob/4b679fddefe2bc1d34e2f14eb9ef171c46d6bffe/src/applications/personalization/profile/sass/profile-mobile-subnav.scss, but the markup for that menu is different/more complex than this simple button. In the file I linked to, the button looks identical to this button, but functions entirely differently so the markup (and thus CSS) is different.

Comment (erikphansen) [reply packages/formation/sass/modules/_m-button.scss:151]: Also, styles similar to this (for the Pittsburgh VA's menu) live outside of formation in https://github.com/department-of-veterans-affairs/vets-website/blob/cbfe8334fd5a2d2db7bf635b916cfa808a9670fc/src/platform/site-wide/sass/modules/_m-side-nav.scss

**Pull request merged**
